spark/src/main/scala/io/substrait/spark/ToSubstraitType.scala

```diff
@@ -122,52 +163,35 @@ class ToSubstraitType {
             keyT =>
               convert(valueType, Seq.empty, valueContainsNull)
                 .map(valueT => creator.map(keyT, valueT)))
-      case _ =>
-        None
+      case StructType(fields) =>
+        if (fields.isEmpty) {
+          Some(creator.struct(JavaConverters.asJavaIterable(Seq.empty)))
+        } else {
+          Util
+            .seqToOption(fields.map(f => convert(f.dataType, f.nullable)))
+            .map(l => creator.struct(JavaConverters.asJavaIterable(l)))
+        }
+      case _ => None
     }
   }
-  def toNamedStruct(output: Seq[Attribute]): Option[NamedStruct] = {
-    val names = JavaConverters.seqAsJavaList(output.map(_.name))
-    val creator = Type.withNullability(false)
-    Util
-      .seqToOption(output.map(a => convert(a.dataType, a.nullable)))
-      .map(l => creator.struct(JavaConverters.asJavaIterable(l)))
-      .map(NamedStruct.of(names, _))
-  }
-  def toNamedStruct(schema: StructType): NamedStruct = {
-    val creator = Type.withNullability(false)
-    val names = new java.util.ArrayList[String]
-    val children = new java.util.ArrayList[Type]
-    schema.fields.foreach(
-      field => {
-        names.add(field.name)
-        children.add(apply(field.dataType, field.nullable))
-      })
-    val struct = creator.struct(children)
-    NamedStruct.of(names, struct)
-  }
 
-  def toStructType(namedStruct: NamedStruct): StructType = {
```

Comment:
these were previously inside `ToSubstraitType`, which is confusing given they return Spark types. Now they're moved into the ToSparkType object above, but also rewritten to use the newly-available struct type code.

(We should also split this file, but I'd rather do that as FLUP to keep the diff clean)